src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -246,48 +247,61 @@ def _accumulate_energy(self, area, child, is_cep: bool):
                     if trade.buyer is child.name:
                         self._total_energy += trade.offer.energy
 
-    def _export_house_pv_self_consumption(self, area):
-        house_pv_device = list()
-        house_load_device = list()
-        trade_by_pv = 0
-        total_energy_bought = 0
-        traded_from_pv = 0
-        if not _is_house_node(area):
-            return
+    def _export_area_sufficiency(self, area):
+        area_load_device = list()
+        area_ess_device = list()
+        area_producer_device = list()
+        self_charged_ess_energy = 0  # charge-> + ; discharge-> -
+
         for child in area.children:
-            if isinstance(child.strategy, PVStrategy):
-                house_pv_device.append(child.name)
-            elif _is_load_node(child) or _is_prosumer_node(child):
-                house_load_device.append(child.name)
+            if _is_load_node(child):
+                area_load_device.append(child.name)
+            elif _is_prosumer_node(child):
+                area_ess_device.append(child.name)
+            elif _is_producer_node(child):
+                area_producer_device.append(child.name)
+            else:
+                area_load_device.append(make_iaa_name(child))
```

Comment:
This block is for the storage?

Reply:
Sorry, no. Its further up.
But for what case is this block?

Reply:
Maybe this handles the Area class?

Reply:
Yes, this block is to handle the Area class

Reply:
@mfatma, @sarahhambridge, @spyrostzand @hannesdiedrich. Can you guys give me little clarity to either consider area as load, producer or prosumer device?

Reply:
An area should be a prosumer, right?